### PR TITLE
Setup for Prow

### PR DESCRIPTION
**Pull request by rashmigottipati**
**Description of the change:**

- Add `Dockerfile`, `ci-operator.yaml` file 
- Update Makefile to use the `build/` directory 
- Create an openshift directory and in that directory, add a Makefile that includes the targets/openshift/deps.mk helper
- Run go get github.com/openshift/build-machinery-go
- Along with the Makefile, run `go mod init` and add dependencies.go 
- Run `go mod vendor`

**Motivation for the change:**

- Configure prow for the downstream repo

### CODE REVIEW

**Comment by rashmigottipati** [comment]
cc @everettraven @oceanc80

**Comment by everettraven** [comment]
While we don't have commitchecker set up here yet, I do think that it makes sense to try and follow the commit flow we have taken when doing downstream specific changes for the OLMv1 components. I think in general that looks like a set of commits like:
- `UPSTREAM: <carry>: Add OpenShift specific files`
    - This commit I believe should contain all the new files necessary for adhering to the OpenShift and ART requirements
- `UPSTREAM: <drop>: go mod vendor`
    - This commit should contain all the vendored files from running `go mod vendor`

File: openshift/vendor/github.com/openshift/build-machinery-go/.ci-operator.yaml
Comment (everettraven) [inline openshift/vendor/github.com/openshift/build-machinery-go/.ci-operator.yaml:1]: Looking at a similar commit in catalogd, it looks like even the openshift vendor files are included in the drop commit: https://github.com/openshift/operator-framework-catalogd/commit/f8759bb4c432f7f3fbef4a2e6358262eef105be4

Comment (rashmigottipati) [reply openshift/vendor/github.com/openshift/build-machinery-go/.ci-operator.yaml:1]: Thanks for noticing it. Fixed it and amended the last commit to include the openshift vendor files.

File: .ci-operator.yaml
Comment (ncdc) [inline .ci-operator.yaml:4]: @stevekuznetsov should this be rhel 8 or 9, do you know?

**Review by everettraven — APPROVED** [review]
LGTM

File: .ci-operator.yaml
Comment (stevekuznetsov) [reply .ci-operator.yaml:4]: Unless there's a good reason, RHEL9 please. The automation will push us in that direction and getting ahead of the curve helps.

Comment (ncdc) [reply .ci-operator.yaml:4]: @rashmigottipati please make the change to 9 - thanks!

**Review by ncdc — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (ncdc) [inline Dockerfile:1]: I would put this in the openshift dir.

**Comment by ncdc** [comment]
Is it possible to put the `release` dir under `openshift`?

File: .ci-operator.yaml
Comment (rashmigottipati) [reply .ci-operator.yaml]: Updated to RHEL9

File: Makefile
Comment (rashmigottipati) [reply Makefile]: Sounds good, reverted this change

**Comment by rashmigottipati** [comment]
@ncdc I think it's ok to put the `release` dir under `openshift. 
Originally, I followed the same convention as downstream sdk: https://github.com/openshift/ocp-release-operator-sdk/tree/master/release/ansible which has a separate release directory and since this repo is just for ansible without helm and sdk, I think it should be fine.

Moved it under `openshift` dir.

**Comment by grokspawn** [comment]
Not sure if this is to be covered in a follow-on PR, but since this clone from upstream has non-OCP folks in its approval lists, we should create DOWNSTREAM_OWNERS/DOWNSTREAM_OWNERS_ALIASES (like [operator-framework-operator-controller](https://github.com/openshift/operator-framework-operator-controller/blob/main/DOWNSTREAM_OWNERS_ALIASES)) which is the same as in the existing files (for e.g. [OWNERS_ALIASES](https://github.com/openshift/ansible-operator-plugins/blob/main/OWNERS_ALIASES)) except with non-openshift folks filtered out. 

When you do the prow PR you can set up the [_plugins.yaml](https://github.com/openshift/release/blob/ca8e034ced5fd4931175774f5eb0ba5b53d1c379/core-services/prow/02_config/_plugins.yaml#L709) which will then start only listening to folks in the new DOWNSTREAM_OWNER* files.

**Comment by grokspawn** [comment]
Are we going to pursue a rename of this repo to `openshift/operator-framework-ansible-operator-plugins`?

**Comment by rashmigottipati** [comment]
@grokspawn yes, we are going to pursue a rename of the repo to `openshift/operator-framework-ansible-operator-plugins`. 

@oceanc80 was going to try to open a ticket to see if dptp can rename the repo. I think we need a follow-up PR when the rename is done to change the references. IMO, I don't think we should block this PR until then.

File: DOWNSTREAM_OWNERS
Comment (grokspawn) [inline DOWNSTREAM_OWNERS:1]: As we mentioned in slack, please restore the OWNERS.* files, and create new DOWNSTREAM_OWNERS.* files, copying the upstream owners info but omitting any non-RH folks.

File: testdata/memcached-molecule-operator/config/manager/kustomization.yaml
Comment (grokspawn) [inline testdata/memcached-molecule-operator/config/manager/kustomization.yaml:1]: Is this a NOOP that somehow got caught up in the diff oddly?

Comment (rashmigottipati) [reply testdata/memcached-molecule-operator/config/manager/kustomization.yaml:1]: Yeah it was added during my testing and it's not required, I removed it

File: DOWNSTREAM_OWNERS
Comment (rashmigottipati) [reply DOWNSTREAM_OWNERS:1]: Done

File: testdata/memcached-molecule-operator/config/manager/kustomization.yaml
Comment (rashmigottipati) [reply testdata/memcached-molecule-operator/config/manager/kustomization.yaml:1]: Didn't realize a newline was removed, I thought you were referring to the old change on this file. :sweat_smile:  I fixed it now.

**Pull request merged**
